include/aspect/simulator.h

```diff
@@ -1223,6 +1226,13 @@ namespace aspect
        */
       std_cxx11::shared_ptr<MeltHandler<dim> > melt_handler;
 
+      /**
+       * Shared pointer for an instance of the NewtonHandler. This way,
+       * if we do not need the machinery for doing Newton stuff, we do
+       * not even allocate it.
+       */
+      std_cxx11::shared_ptr<NewtonHandler<dim> > newton_handler;
```

Comment:
All of ASPECT right now uses `shared_ptr` -- we didn't have `std_cxx11::unique_ptr` when we started using these pointers. There will have to be a point where we replace many of these by `unique_ptr`.